s3cmd

```diff
@@ -583,12 +594,79 @@ def cmd_object_del(args):
 
         if not recursive:
             rc = subcmd_object_del_uri(uri_str)
-        else:
+        elif Config().exclude:
             rc = subcmd_batch_del(uri_str = uri_str)
+        else:
+            rc = subcmd_batch_del_iterative(uri_str = uri_str)
         if not rc:
             return rc
     return EX_OK
 
+def subcmd_batch_del_iterative(uri_str = None, bucket = None):
+    """ Streaming version of batch deletion (doesn't realize whole list in memory before deleting). 
+
+    Differences from subcmd_batch_del:
+      - Obeys cfg.max_delete, but if it is >1000, will delete files until we are sure that at least 
+        that many files are to be deleted
+      - Only prints out individual file names if <1000 files are to be deleted
```

Comment:
I'm not comfortable at all with this behavior.
Things should be consistent and not based on an arbitrary limit.
In my opinion, "subcmd_batch_del_iterative" should output things in the same way as "subcmd_batch_del" do: print each files that is removed.